test/rules/avoid_unused_constructor_parameters.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_unused_constructor_parameters`
```

Comment:
Just for completeness, it would be good to include a test for a redirecting constructor (`this(foo)`) and a factory constructor (`= Impl(foo)`).